eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MutableMapIterable.java

```diff
@@ -194,6 +194,11 @@ default V add(Pair<? extends K, ? extends V> keyValuePair)
      */
     MutableMapIterable<K, V> newEmpty();
 
+    /**
+     * Creates a new instance of the same type, using the initial capacity.
+     */
+    <K, V> MutableMapIterable<K, V> newEmpty(int initialCapacity);
```

Comment:
Why does this take K and V? I see two problems.

* These are the same names as the outer generic types, which shadows them and is super confusing.
* This cannot work for subclasses that take `Comparator<K>` or `HashingStrategy<K>` in their constructors.